pkg/apis/serving/v1beta1/configuration_validation.go

```diff
@@ -61,3 +65,34 @@ func (cs *ConfigurationStatus) Validate(ctx context.Context) *apis.FieldError {
 func (csf *ConfigurationStatusFields) Validate(ctx context.Context) *apis.FieldError {
 	return nil
 }
+
+// ValidateLabels function validates service labels
+func (c *Configuration) ValidateLabels() (errs *apis.FieldError) {
+	for key, val := range c.GetLabels() {
+		switch {
+		case key == config.VisibilityLabelKey:
+			errs = errs.Also(validateClusterVisbilityLabel(val))
+		case key == serving.RouteLabelKey:
+		case key == serving.ServiceLabelKey:
+			errs = errs.Also(verifyLabelOwnerRef(val, serving.ServiceLabelKey, "Service", c.GetOwnerReferences()))
+		case strings.HasPrefix(key, serving.GroupNamePrefix):
+			errs = errs.Also(apis.ErrInvalidKeyName(key, ""))
+		}
+	}
+	return
+}
+
+// verifyServiceLabelOwnerRef function verifies the owner references of resource with label key has val value.
+func verifyLabelOwnerRef(val, label, resource string, ownerRefs []metav1.OwnerReference) (errs *apis.FieldError) {
+	found := false
+	for i, ref := range ownerRefs {
+		if ref.Kind != resource || val != ref.Name {
+			errs = errs.Also(apis.ErrInvalidArrayValue(ref.Name, label, i))
+		}
+		found = true
```

Comment:
I am not sure this loop is doing what you expect... With this we always return found true on the first item??

Reply:
`found` variable is used for tracking there is at least one owner reference with expected key.

Reply:
@vagababov had a good point that there is no need to enforce. So I will update this logic

Reply:
but you set it to true regardless of whether you found proper one or not.

Reply:
My bad. gotcha

Reply:
I fixed the logic for this function. Please take another look.